### PR TITLE
MultilineBlockBracketsOnSameColumn: Put inherit keyword in record into newline

### LINKED ISSUE #852: MultilineBlockBracketsOnSameColumn not working properly when calling base constructors
Issue created from [fantomas-online](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAXAngB0WAyogE4BuxAwgIYB21s6AQogEqIDOsANmfAKIAe0RNnQBLWNTABeADqSIYUdQAWxUejDlYAW20BXaqOiUxEgKrU2e6ELZsAZns4ChI8dTlyFYaogDuABTa7GyUAOaIAJTSXt4KwIoqahpaugZGJu4WVjYhjs6CwqbUQSHhUWAAvpAANOBQbIjoYtRhbKhgANr1EKBxUAAkWLgdkErotT0KkAMklJx6yGjd-RAADHWrUACS1EjU6PjYlEIAcnrak1sALFMQALpTlZvxd4PDS1DjV3EzcwufFarACML36kAACuUAOqieDoZQ-frAgBMazej28zymfV%2BQxwn0gACNYFwkdNZvNFh0gf0UWDfoRtEYuBIAILoXh7ADy9gAMkpkAzvPZ5o0MU8GbjvDMPqMSWThYN-tTlm8IABmJUQSBHE6IJj2WBERBQoiUYLoYjk7zoIiLCVYqVvWUE%2BWkzg25VUwHqsDXbVQPVCQ3GxB82B%2BYjGRq7EiwYzFL0QO0O-qYhTY7zSilytDEj1ev4%2Bml%2BgCsgd1xxDiCNJrM2FwRBjiDjCcyEmTYFFnHF6clOJd%2BJG%2BYVnqVxYBpa2ADZK8GDbWw%2BROJQ7FpLKnoOhjV2e324hmIFnXuDh4Sx0XKVO1VsAOzz6uLuuIACyiG0ROtgf3iEdmedM88ygS8J2vVUuj9AAOR99VDE0tE4TsfzFP9%2BydQcgLdUdCzAlVfS2ABOWChDZewrSIVJtEoLtUzQw8B2zIdgILRUh3w6cQQ2P0qzgpcTSZFkkOoPdUP-Y9ALxFjQPYktbxBUEeIXMiKMEuBhNo%2B16O8I8qkkmVz3dNizw4%2BTkXpHjdn2dBuWoAAVIhMFhBFRN7bSFF0k9emY7CQNw2Sb0grZgS1JSnzZIhYAMeAABFEE4URmQozS0wYjCmKwkcvgOK9TKCkEAx419KH4bZ7Ds1RqF4Nz8GUY10FheFEUDa50XQgDMKk3yxhyvC5Py5EKyKkrdnsUR%2BG5JsTGNARsBNOx3C7Ms2rSjqMq6rKeomPrAtpOJgTnYb%2BFYOAiHgRqXJaladMY08NsJb4dogvbvGBB8joi81MG5IgBTYBq4Uuv1WvEvTOoMljHoC56-WBGCjr5JoGCUeAlDCC7muB66PNu7zMovfyTP6l6FGBYiiqcMQEt8BgkOgABrBhzQZpo2Fs-ALUQRCLhEwM6NBrywBzHVDJw4y8TykmIDRStTn8anF3QKNEHsgk4rGwxijZPZ30-Yh2hQtyBf03Nupkondr9FFFK2SAAGlEGEMqKpV3YOeCAVfFcg8bvSu6IbNwmJeJq2LNt-A7SMdBXwQIU-V-AX6nuF4xjYAAxNhRA6BPKiAA)

**Please describe here fantomas problem you encountered**

#### Code

```fsharp
type ServerCannotBeResolvedException =
    inherit CommunicationUnsuccessfulException

    new(message) =
        { inherit CommunicationUnsuccessfulException(message) }
```
            

#### Result

With MultilineBlockBracketsOnSameColumn disabled:

```fsharp
type ServerCannotBeResolvedException =
    inherit CommunicationUnsuccessfulException

    new(message) =
        { inherit CommunicationUnsuccessfulException(message) }
```

With MultilineBlockBracketsOnSameColumn enabled:
```fsharp
type ServerCannotBeResolvedException =
    inherit CommunicationUnsuccessfulException

    new(message) =
        { inherit CommunicationUnsuccessfulException(message)

        }
```

Expected results (with MultilineBlockBracketsOnSameColumn enabled):

```fsharp
type ServerCannotBeResolvedException =
    inherit CommunicationUnsuccessfulException

    new(message) =
        {
            inherit CommunicationUnsuccessfulException(message)
        }
```

#### Options

Fantomas Next - 4.0.0-alpha-001-1/1/1990

| Name | Value |
| ---- | ----- |
| `IndentSpaceNum` | `4` |
| `PageWidth` | `120` |
| `SemicolonAtEndOfLine` | `false` |
| `SpaceBeforeParameter` | `true` |
| `SpaceBeforeLowercaseInvocation` | `true` |
| `SpaceBeforeUppercaseInvocation` | `false` |
| `SpaceBeforeClassConstructor` | `false` |
| `SpaceBeforeMember` | `false` |
| `SpaceBeforeColon` | `false` |
| `SpaceAfterComma` | `true` |
| `SpaceBeforeSemicolon` | `false` |
| `SpaceAfterSemicolon` | `true` |
| `IndentOnTryWith` | `false` |
| `SpaceAroundDelimiter` | `true` |
| `MaxIfThenElseShortWidth` | `40` |
| `MaxInfixOperatorExpression` | `50` |
| `MaxRecordWidth` | `40` |
| `MaxArrayOrListWidth` | `40` |
| `MaxLetBindingWidth` | `40` |
| **`MultilineBlockBracketsOnSameColumn`** | **`true`** |
| `NewlineBetweenTypeDefinitionAndMembers` | `false` |
| `KeepIfThenInSameLine` | `false` |
| `StrictMode` | `false` |

### PR DESCRIPTION
**Pull request by Bobface**
Fixes #852 

Puts the `inherit`-keyword into a new line and indents when `MultilineBlockBracketsOnSameColumn` is enabled. Example:

**Code**
```
let a =
    { inherit ProjectPropertiesBase<_>(projectTypeGuids, factoryGuid, targetFrameworkIds, dotNetCoreSDK)
      buildSettings = FSharpBuildSettings()
      targetPlatformData = targetPlatformData }
```
**Result**
```
let a =
    {
        inherit ProjectPropertiesBase<_>(projectTypeGuids, factoryGuid, targetFrameworkIds, dotNetCoreSDK)
        buildSettings = FSharpBuildSettings()
        targetPlatformData = targetPlatformData
    }
```

At the time of writing this it is not entirely clear if the behaviour this PR implements is intended. Thus it's a draft for now until we know more.

### CODE REVIEW

**Comment by Bobface** [comment]
@nojaf Ready for review. I implemented it as you suggested in #852:

> 
> 
> When empty I would go for:
> 
> ```fsharp
> { inherit CommunicationUnsuccessfulException(message) }
> ```
> 
> when not empty:
> 
> ```fsharp
> {
>     inherit CommunicationUnsuccessfulException(message)
>     SomeA = ThingA
>     SomeB = ThingB
> }
> ```

File: src/Fantomas/CodePrinter.fs [hunk 1]
```diff
@@ -1655,12 +1655,14 @@ and genMultilineRecordInstanceAlignBrackets
     astContext
     =
     let fieldsExpr = col sepSemiNln xs (genRecordFieldName astContext)
+    let hasFields = List.length xs > 0
```
Comment (nojaf) [inline src/Fantomas/CodePrinter.fs:1658]: We has a helper function `List.isNotEmpty` for this

**Comment by Bobface** [comment]
@nojaf Added the test.

**Comment by nojaf** [comment]
@Bobface thanks!

**Pull request merged**